test/extensions/geoip_providers/maxmind/geoip_provider_test.cc

```diff
@@ -620,8 +637,9 @@ TEST_P(MmdbReloadImplTest, MmdbReloadedInFlightReadsNotAffected) {
   std::string reloaded_db_file_path = TestEnvironment::substitute(test_case.reloaded_db_file_path_);
   TestEnvironment::renameFile(source_db_file_path, source_db_file_path + "1");
   TestEnvironment::renameFile(reloaded_db_file_path, source_db_file_path);
+  cb_added_opt.value().waitReady();
   {
-    absl::MutexLock guard(&mutex_);
+    absl::ReaderMutexLock guard(&mutex_);
```

Comment:
why the change to reader/writer locks?

Reply:
We did not need writer lock here, it just reads the value.